src/coordinator/functions/logical/and.go

```diff
@@ -24,77 +24,35 @@ import (
 	"math"
 
 	"github.com/m3db/m3db/src/coordinator/block"
-	"github.com/m3db/m3db/src/coordinator/executor/transform"
-	"github.com/m3db/m3db/src/coordinator/parser"
 )
 
-const (
-	// AndType uses values from left hand side for which there is a value in right hand side with exactly matching label sets.
-	// Other elements are replaced by NaNs. The metric name and values are carried over from the left-hand side.
-	AndType = "and"
-)
-
-// NewAndOp creates a new And operation
-func NewAndOp(lNode parser.NodeID, rNode parser.NodeID, matching *VectorMatching) BaseOp {
-	return BaseOp{
-		OperatorType: AndType,
-		LNode:        lNode,
-		RNode:        rNode,
-		Matching:     matching,
-		ProcessorFn:  NewAndNode,
-	}
-}
-
-// AndNode is a node for And operation
-type AndNode struct {
-	op         BaseOp
-	controller *transform.Controller
-}
-
-// NewAndNode creates a new AndNode
-func NewAndNode(op BaseOp, controller *transform.Controller) Processor {
-	return &AndNode{
-		op:         op,
-		controller: controller,
-	}
-}
-
-// Process processes two logical blocks, performing And operation on them
-func (c *AndNode) Process(lhs, rhs block.Block) (block.Block, error) {
-	lIter, err := lhs.StepIter()
-	if err != nil {
-		return nil, err
-	}
-
-	rIter, err := rhs.StepIter()
-	if err != nil {
-		return nil, err
-	}
+func makeAndBuilder(
```

Comment:
name is misleading since it's not really a builder. Maybe call it makeAndBlock ?

Reply:
Will do, this was a holdover from different approach which I didn't update